test/unit/lib/inject.js

```diff
@@ -30,29 +30,116 @@ describe('lib/inject', function() {
 		});
 	});
 
-	it('should select elements using `hideElements` option', function(done) {
-		options.hideElements = '#ad, .modal';
-		inject(window, options, function() {
-			assert.calledOnce(window.document.querySelectorAll);
-			assert.calledWith(window.document.querySelectorAll, '#ad, .modal');
+	it('should ignore messages when they are a child of `options.hideElements`', function(done) {
+		options.hideElements = '.hide';
+		window.document.querySelectorAll.returns([
+			{
+				outerHTML: '<element class="hide">parent</element>'
+			}
+		]);
+		window.HTMLCS.getMessages.returns([
+			{
+				code: 'foo-code',
+				element: {
+					parentNode: null,
+					isEqualNode: function() {return false;},
```

Comment:
I am not keen on this approach but its the cleanest approach I have found so far.